lib/parser/gpx.rb

```diff
@@ -42,6 +46,40 @@ def parse # rubocop:disable Metrics/AbcSize
 
     private
 
+    def enhance_trace(points)
+      prev_point = points.first
+      points[1..].each_with_object([]) do |cur_point, et|
+        duration = cur_point[:time].to_f - prev_point[:time].to_f
```

Comment:
If the user is not moving, are the GPS traces not recorded?
Or do we receive GPS with the same lat/lng?

Reply:
I would say: depends on the device/implementation. For Garmin it seems it does not record gps points. Runtastic is also behaving like this iirc.